lib/web/users.go

```diff
@@ -305,9 +311,10 @@ type userTraits struct {
 // - if the value is an empty array we remove every element from the trait
 // - otherwise, we replace the list for that trait
 type saveUserRequest struct {
-	Name   string     `json:"name"`
-	Roles  []string   `json:"roles"`
-	Traits userTraits `json:"traits"`
+	Name      string              `json:"name"`
+	Roles     []string            `json:"roles"`
+	Traits    userTraits          `json:"traits"`
+	AllTraits map[string][]string `json:"allTraits"`
 }
 
 func (r *saveUserRequest) checkAndSetDefaults() error {
```

Comment:
`Traits` and `AllTraits` are mutually exclusive it seems like, we should update checkAndSetDefaults accordingly.

Reply:
updated checkAndSetDefaults in https://github.com/gravitational/teleport/pull/42620/commits/0d6b589bcda5f4c5525ffba6964a281b30d7025b